.github/workflows/check.yml

```diff
@@ -8,52 +8,65 @@ env:
 
 jobs:
   check:
-    name: Ubuntu 21.04 build
+    name: Ubuntu Rolling build
     runs-on: ubuntu-latest
-    container: ubuntu:21.04
     strategy:
       matrix:
         compiler: ['gcc', 'clang']
 
     env:
       CC: ${{ matrix.compiler }}
       CFLAGS: -fsanitize=undefined -fsanitize-undefined-trap-on-error -fsanitize=address -Wp,-D_FORTIFY_SOURCE=2
+      BUILD_CONTAINER: ${{ matrix.compiler }}-build-container
+      RUN_CMD: docker exec -t -w /src -e TEST_IN_CI -e ASAN_OPTIONS -e G_MESSAGES_DEBUG -e XDG_DATA_DIRS ${{ matrix.compiler }}-build-container
 
     steps:
+    - name: Prepare container
+      run: |
+        docker run --name $BUILD_CONTAINER \
+          --tty --device /dev/fuse --cap-add SYS_ADMIN \
+          --security-opt apparmor:unconfined \
+          -v $(pwd):/src \
+          -e DEBIAN_FRONTEND \
+          -e DEBCONF_NONINTERACTIVE_SEEN=true \
+          -e TERM=dumb \
+          -e CC -e CFLAGS \
+          -d ubuntu:rolling sleep infinity
```

Comment:
Why does this need to be rolling, I'd prefer staying on versioned releases.

Reply:
Sure, [that's just an alias](https://hub.docker.com/_/ubuntu) for the latest-released release (so 21.10 at the date).

I know the name is misleading, but that's refers to the latest stable, otherwise it would have been `devel` (that is `jammy` 22.04 right now).

Reply:
I prefer just setting 21.10 as it avoids a chance of waking up with it suddenly broken when 22.04 comes out, instead we can choose when to dedicate time to updating it.

Reply:
Ok, fair enough... I've changed it.